solutions/instances/variables.tf

```diff
@@ -0,0 +1,270 @@
+# Cloud Monitoring Variables
+##############################################################################
+
+variable "cloud_monitoring_instance_name" {
+  type        = string
+  description = "The name of the IBM Cloud Monitoring instance to create."
+  default     = "cloud-monitoring"
+}
+
+variable "cloud_monitoring_plan" {
+  type        = string
+  description = "The IBM Cloud Monitoring plan to provision. Available: lite, graduated-tier, graduated-tier-sysdig-secure-plus-monitor"
+  default     = "graduated-tier" # Need to confirm which plan we are using
+
+  validation {
+    condition     = can(regex("^lite$|^graduated-tier$|^graduated-tier-sysdig-secure-plus-monitor$", var.cloud_monitoring_plan))
+    error_message = "The cloud_monitoring_plan value must be one of the following: lite, graduated-tier, graduated-tier-sysdig-secure-plus-monitor."
+  }
+}
+
+variable "cloud_monitoring_tags" {
+  type        = list(string)
+  description = "Tags associated with the IBM Cloud Monitoring instance (Optional, array of strings)."
+  default     = []
+}
+
+variable "cloud_monitoring_service_endpoints" {
+  description = "The type of the service endpoint that will be set for the IBM cloud monitoring instance."
+  type        = string
+  default     = "public-and-private" # Change it to private
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.cloud_monitoring_service_endpoints)
+    error_message = "The specified service_endpoints is not a valid selection"
+  }
+}
+
+########################################################################################################################
+# COS variables
+########################################################################################################################
+
+variable "add_bucket_name_suffix" {
+  type        = bool
+  description = "Add random generated suffix (4 characters long) to the newly provisioned COS bucket name. Only used if not passing existing bucket. set to false if you want full control over bucket naming using the 'cos_bucket_name' & 'cos_target_bucket_name' variable."
+  default     = true
+}
+
+variable "cos_region" {
+  type        = string
+  default     = "us-south"
+  description = "The Cloud Object Storage region."
+}
+
+variable "cos_instance_name" {
+  type        = string
+  default     = "observability-cos"
+  description = "The name to use when creating the Cloud Object Storage instance."
+}
+
+variable "cos_instance_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to Cloud Object Storage instance. Only used if not supplying an existing instance."
+  default     = []
+}
+
+variable "cos_instance_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the Cloud Object Storage instance. Only used if not supplying an existing instance."
+  default     = []
+}
+
+variable "cos_bucket_name" {
```

Comment:
Just so its clear what this bucket is used for, suggest renaming the variable to `log_archive_cos_bucket_name`